packages/fhir-converter/src/lib/outputProcessor/resourceDeduplicator.js

```diff
@@ -0,0 +1,98 @@
+//
+//     THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+//     WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+//     WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+//     AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+//     CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+//     OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+//     NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+//     CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+// -------------------------------------------------------------------------------------------------
+
+var merge = require("./resourceMerger").merge;
+
+module.exports.Process = function (jsonObj) {
+  try {
+    const encounters = jsonObj.entry.filter(entry => {
+      return entry.resource.resourceType === "Encounter";
+    });
+
+    const idMap = new Set();
+    let duplicateId;
+    encounters.forEach(enc => {
+      enc.resource.identifier.forEach(id => {
+        if (idMap.has(id.value)) {
+          duplicateId = id.value;
+        } else {
+          idMap.add(id.value);
+        }
+      });
+    });
+
+    if (duplicateId) {
+      const duplicateEncounters = encounters.filter(enc => {
+        return enc.resource.identifier.some(id => id.value === duplicateId);
+      });
+      const enc1 = duplicateEncounters[0];
+      const enc2 = duplicateEncounters[1];
+      const mergedEncounter = merge(enc1, enc2);
+
+      const deduplicatedIdentifiers = deduplicateIdentifiers(mergedEncounter.resource.identifier);
+      mergedEncounter.resource.identifier = deduplicatedIdentifiers;
+      mergedEncounter.status = selectMostInformativeStatus(enc1.status, enc2.status);
+
+      const entriesWithoutEncounters = jsonObj.entry.filter(entry => {
+        const includes = [enc1.resource.id, enc2.resource.id].includes(entry.resource.id);
+        return !includes;
+      });
+
+      jsonObj.entry = [...entriesWithoutEncounters, mergedEncounter];
+      jsonObj.entry = JSON.parse(
+        JSON.stringify(jsonObj.entry).replaceAll(enc1.resource.id, mergedEncounter.resource.id)
+      );
+    }
+    return jsonObj;
+  } catch (err) {
+    return jsonObj;
+  }
+};
+
+function deduplicateIdentifiers(identifiers) {
+  const identifierMap = {};
+
+  identifiers.forEach(identifier => {
+    const key = `${identifier.system}-${identifier.value}`;
+    if (!identifierMap[key]) {
+      identifierMap[key] = identifier;
+    } else {
+      const existingFields = Object.keys(identifierMap[key]).length;
+      const newFields = Object.keys(identifier).length;
+      if (newFields > existingFields) {
+        identifierMap[key] = identifier;
+      }
+    }
+  });
+
+  return Object.values(identifierMap);
+}
+
+function selectMostInformativeStatus(status1, status2) {
+  if (status1 === status2) {
+    return status1;
+  }
+  return [status1, status2].filter(status => status !== "unknown")[0] || "unknown";
```

Comment:
What if one status is "finished" and one status is "started"? This handles one possible status and then defaults to the first one. 

How many possible status' are there? Maybe it's not to complex to define a ranking over all of them if there are like 5

Reply:
I ran a script on this, and the encompassing encounter only ever has `unknown`, and the other one is always `finished`. Literally, the map looked like: 
```
"encompassingEncounter": { "unknown": 89xx }, 
"internalEncounter": { "finished": 89xx }
```

Reply:
Ok fair but if you look at the CDA spec and there like only one or two more options, nbd to add those. Just saying that out test set isn't always representative :)

Reply:
Well, in that case, we can't make the call on which one is "more correct". So what do you suggest we do?